MINOR: Remove RaftManager.maybeDeleteMetadataLogDir and AutoTopicCreationManagerTest.scala

cmccabe opened the pull request:
Remove RaftManager.maybeDeleteMetadataLogDir since it's not used any more. also remove AutoTopicCreationManagerTest since it tests the ZK-mode-only AutoTopicReationManager.

chia7712 commented on core/src/test/scala/integration/kafka/server/DynamicBrokerReconfigurationTest.scala:
```diff
@@ -1037,237 +923,6 @@ class DynamicBrokerReconfigurationTest extends QuorumTestHarness with SaslSetup
   }
 
-  @Test
-  // Modifying advertised listeners is not supported in KRaft
```
Out of curiosity, do we have plans to support this for KRaft? I saw there's a JIRA about revising the docs (https://issues.apache.org/jira/browse/KAFKA-16617)

chia7712 commented on core/src/test/scala/integration/kafka/server/DynamicBrokerReconfigurationTest.scala:
```diff
@@ -1037,237 +923,6 @@ class DynamicBrokerReconfigurationTest extends QuorumTestHarness with SaslSetup
-  }
-
-  @ParameterizedTest
-  @ValueSource(strings = Array("zk", "kraft"))
```
Apologies, but why are we removing the entire test case instead of keeping the KRaft-related parts?

chia7712 reviewed (commented):
@cmccabe could you please check the failed test? `testDefaultTopicConfig`, `testMetricsReporterUpdate`, and `testThreadPoolResize` need to add `@ParameterizedTest` and `@ValueSource(strings = Array("kraft"))` to ensure they are running with kraft.

mimaison commented:
@cmccabe Can you rebase to resolve the conflicts? Thanks

cmccabe committed:
KAFKA-17615: Remove KafkaServer references (#17365)

Remove ReplicationQuotaUtils.scala since it's not used any more.

Remove DynamicBrokerConfigTest.testEnableDefaultUncleanLeaderElection
since it tests the ZK-based controller, and we have other tests for
configuring unclean leader election in KRaft mode.

Remove RaftManager.maybeDeleteMetadataLogDir since it was only used
during ZK migration, and that code has been removed. Similarly, remove
RaftManagerTest.testKRaftBrokerDoesNotDeleteMetadataLog which tested
that function.

Move KafkaServer.MIN_INCREMENTAL_FETCH_SESSION_EVICTION_MS to KafkaBroker.

Remove AutoTopicCreationManagerTest since it tests the ZK-mode-only
AutoTopicReationManager.

cmccabe commented:
@mimaison : just rebased.

mimaison reviewed (approved):
LGTM!

mimaison commented:
The test failure is https://issues.apache.org/jira/browse/KAFKA-18297, so it's not related to this PR.

chia7712 committed:
Merge branch 'trunk' into KAFKA-17615
